cpp/dolfinx/fem/Function.h

```diff
@@ -153,39 +153,21 @@ class Function
   /// @brief Interpolate a provided Function.
   /// @param[in] v The function to be interpolated
   /// @param[in] cells The cells to interpolate on
-  /// @param[in] cell_map A map from cells in the mesh associated with `this`
-  /// function to cells in mesh associated with `v`
-  /// @param[in] nmm_interpolation_data Auxiliary data to interpolate on
-  /// nonmatching meshes. This data can be generated with
-  /// generate_nonmatching_meshes_interpolation_data (optional).
-  void interpolate(
-      const Function<value_type, geometry_type>& v,
-      std::span<const std::int32_t> cells,
-      std::span<const std::int32_t> cell_map,
-      const std::tuple<std::span<const std::int32_t>,
-                       std::span<const std::int32_t>,
-                       std::span<const geometry_type>,
-                       std::span<const std::int32_t>>& nmm_interpolation_data
-      = {})
+  /// @param[in] cell_map A map from cells in the mesh associated with \p this
+  /// function to cells in mesh associated with \p v
+  void interpolate(const Function<value_type, geometry_type>& v,
+                   std::span<const std::int32_t> cells,
+                   std::span<const std::int32_t> cell_map)
   {
-    fem::interpolate(*this, v, cells, cell_map, nmm_interpolation_data);
+    fem::interpolate(*this, v, cells, cell_map);
   }
 
   /// @brief Interpolate a provided Function.
   /// @param[in] v The function to be interpolated
-  /// @param[in] cell_map Map from cells in self to cell indices in `v`
-  /// @param[in] nmm_interpolation_data Auxiliary data to interpolate on
-  /// nonmatching meshes. This data can be generated with
-  /// generate_nonmatching_meshes_interpolation_data (optional).
-  void interpolate(
-      const Function<value_type, geometry_type>& v,
-      std::span<const std::int32_t> cell_map = std::span<const std::int32_t>()
-      = {},
-      const std::tuple<std::span<const std::int32_t>,
-                       std::span<const std::int32_t>,
-                       std::span<const geometry_type>,
-                       std::span<const std::int32_t>>& nmm_interpolation_data
-      = {})
+  /// @param[in] cell_map Map from cells in self to cell indices in \p v
+  void interpolate(const Function<value_type, geometry_type>& v,
+                   std::span<const std::int32_t> cell_map
```

Comment:
The double `std::span<const std::int32_t>()` looks weird.

Reply:
You suggested that I should add this back in: https://github.com/FEniCS/dolfinx/pull/3114#discussion_r1555708444

Reply:
The double equality in the comment is a typo.

Reply:
So you would like `std::span<const std::int32_t> cell_map = {}` ?